src/include/txn.i

```diff
@@ -867,16 +868,18 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
     /*
      * If the stop pair is set, that means that there is a tombstone at that time. If the stop time
      * pair is visible to our txn then that means we've just spotted a tombstone and should return
-     * "not found", except for history store scan during rollback to stable.
+     * "not found", except when we are doing history store scans during rollback to stable and when
+     * we are told to ignore non-globally visible tombstones.
      */
     if (stop.txnid != WT_TXN_MAX && stop.timestamp != WT_TS_MAX &&
-      (!WT_IS_HS(S2BT(session)) || !F_ISSET(session, WT_SESSION_IGNORE_HS_TOMBSTONE)) &&
-      __wt_txn_visible(session, stop.txnid, stop.timestamp)) {
+      __wt_txn_visible(session, stop.txnid, stop.timestamp) &&
+      ((!F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) &&
+         (!WT_IS_HS(S2BT(session)) || !F_ISSET(session, WT_SESSION_ROLLBACK_TO_STABLE))) ||
+          __wt_txn_visible_all(session, stop.txnid, stop.timestamp))) {
         cbt->upd_value->buf.data = NULL;
         cbt->upd_value->buf.size = 0;
         cbt->upd_value->start_ts = stop.timestamp;
         cbt->upd_value->txnid = stop.txnid;
-        cbt->upd_value->type = WT_UPDATE_TOMBSTONE;
```

Comment:
I didn't understand the reason behind this change, can you explain it?

Reply:
Oh, that's by accident.

Reply:
fixed